test/integration/react-18/test/require-hook.js

```diff
@@ -0,0 +1,16 @@
+const mod = require('module')
+
+const hookPropertyMap = new Map(
+  [
+    ['react', 'react-18'],
+    ['react-dom', 'react-dom-18'],
+    ['react-dom/server', 'react-dom-18/server'],
+  ].map(([request, replacement]) => [request, require.resolve(replacement)])
+)
```

Comment:
```suggestion
const hookPropertyMap = new Map([
  ['react', 'react-18'],
  ['react-dom', 'react-dom-18'],
  ['react-dom/server', 'react-dom-18/server'],
])
```

I think you could omit that resolving (even while it doesn't hurt) as you are rewriting the `request` and it will still going to be resolved afterwards.